packages/atomic/src/components/search/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx

```diff
@@ -0,0 +1,189 @@
+import {Component, h, Listen, State} from '@stencil/core';
+import ArrowRightIcon from 'coveo-styleguide/resources/icons/svg/arrow-right-rounded.svg';
+import ArrowLeftIcon from 'coveo-styleguide/resources/icons/svg/arrow-left-rounded.svg';
+import {Button} from '../../common/button';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+import {
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {Hidden} from '../../common/hidden';
+
+type ArrowDirection = 'right' | 'left';
+
+/**
+ * @internal
+ * The 'atomic-segmented-facet-scrollable' component wraps around one or several 'atomic-segmented-facet' to provide horizontal scrolling capabilities.
+ * @part scrollable-container - The wrapper for the entire component including the horizontal-scroll container and the arrow buttons.
+ * @part horizontal-scroll - The scrollable container for the segmented facets.
+ * @part left-arrow-box - The left arrow box containing both the left arrow button and the fade.
+ * @part right-arrow-box - The right arrow box containing both the right arrow button and the fade.
+ * @part arrow-button - The arrow button used to scroll left or right.
+ * @part fade - The white to transparent gradient.
+ */
+
+@Component({
+  tag: 'atomic-segmented-facet-scrollable',
+  styleUrl: 'atomic-segmented-facet-scrollable.pcss',
+  shadow: true,
+})
+export class AtomicSegmentedFacetScrollable implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
+  private horizontalScroll?: HTMLDivElement;
+  @State() private hideLeftArrow = true;
+  @State() private hideRightArrow = false;
+  private ro!: ResizeObserver;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    console.log('init');
+  }
+
+  componentDidLoad() {
+    this.ro = new ResizeObserver(() => {
+      this.handleScroll();
+    });
+    this.ro.observe(this.horizontalScroll as HTMLDivElement);
+  }
+
+  disconnectedCallback() {
+    this.ro.disconnect();
+  }
+
+  @Listen('mousewheel')
+  handleScroll() {
+    if (!this.horizontalScroll) {
+      return;
+    }
+
+    const isOverflowing =
```

Comment:
a suggestion for the name: `isScrollable`

Reply:
Sure !